src/olympia/devhub/tasks.py

```diff
@@ -180,11 +180,12 @@ def wrapper(task, results, pk, *args, **kwargs):
                 results, type_='error',
                 message=exc.message, msg_id='unsupported_filetype')
             return results
-        except BadZipfile:
+        except BadZipFile:
+            # If we raised a BadZipFile we can return a single exception with
+            # a generic message indicating the zip is invalid or corrupt.
             results = deepcopy(amo.VALIDATOR_SKELETON_EXCEPTION_WEBEXT)
-            annotations.insert_validation_message(
-                results, type_='error',
-                message=ugettext('Invalid or corrupt add-on file.'))
+            results['messages'][0]['message'] = ugettext(
+                'Invalid or corrupt add-on file.')
```

Comment:
Why can't we use annotations here any longer? (it's still used a few lines above)

Reply:
Because I don't want to insert a new error, I want to replace the base one. 

The main difference with before is that before, we ended up catching some invalid zip files later, when validating the files inside, and we raised a `ValidationError`. That was turned by this decorator into a generic linter error using the code just below this:
```python
        except Exception as exc:
            log.exception('Unhandled error during validation: %r' % exc)
            results = deepcopy(amo.VALIDATOR_SKELETON_EXCEPTION_WEBEXT)
            return results
```

Note that in that case, there is only one error returned. 

Now, with Python 3.8, `BadZipFile` is raised earlier than before for the same invalid zips. So we end up in that `except` clause, where we previously returned *two* errors: `Invalid or corrupt add-on file` *and* the generic ` coming from `VALIDATOR_SKELETON_EXCEPTION_WEBEXT`, `Sorry, we couldn't load your WebExtension.`. To stay consistent with what we were doing before for those zips, I'm changing the way we deal with `BadZipFile` and only return *one* error in that case, the `Invalid or corrupt add-on file`. This means I can no longer use `insert_validation_message`, as that would add a new error message instead of replacing the generic one.